aten/src/ATen/cpu/vec/vec256/vec256_complex_double.h

```diff
@@ -140,14 +135,15 @@ template <> class Vectorized<c10::complex<double>> {
     return _mm256_and_pd(angle, real_mask);         // angle    0
   }
   Vectorized<c10::complex<double>> sgn() const {
-    auto abs = abs_();
+    auto abs_zero = abs().values;             // abs 0
+    auto abs = _mm256_movedup_pd(abs_zero);  // abs abs
```

Comment:
You can reduce the number of mask and shuffle ops by just changing `abs_` to:
```
  __m256d abs_() const {
    auto real = _mm256_movedup_pd(values);  // real real
    auto imag = _mm256_movehdup_pd(values); // imag imag
    return Sleef_hypotd4_u05(real, imag);   // abs  abs
  }
```

Reply:
and with this change only `abs_()` has to be modified.

Reply:
Oh, great point! I'll do that.